src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -189,12 +189,29 @@ public void InitializeService(ServiceHost serviceHost)
             try
             {
                 FileBrowserOperation browser;
+                ConnectionInfo connInfo;
                 result.Succeeded = ownerToFileBrowserMap.TryGetValue(fileBrowserParams.OwnerUri, out browser);
-                if (result.Succeeded && browser != null)
+                this.ConnectionServiceInstance.TryFindConnection(fileBrowserParams.OwnerUri, out connInfo);
+
+                if (result.Succeeded && browser != null && connInfo != null)
                 {
-                    result.Children = browser.GetChildren(fileBrowserParams.ExpandPath).ToArray();
-                    result.ExpandPath = fileBrowserParams.ExpandPath;
-                    result.OwnerUri = fileBrowserParams.OwnerUri;
+                    QueueItem queueItem = expandNodeQueue.QueueBindingOperation(
```

Comment:
this will block the restore operation. If you open the file browser once the connection stays open in the queue and restore cannot work after that. If you want to the binding queue please register it in the connection service the same way oe did so the connection get closed temporary when restore is working